src/DataLib/fendl32B_retrofit/gendf_tools.py

```diff
@@ -0,0 +1,201 @@
+# Import packages
+import argparse
+import csv
+import sys
+from logging_config import logger, LoggerWriter
+import ENDFtk
+import contextlib
+import os
+
+# Define an argument parser
+def fendl_args():
+    # Temporarily reset stdout and stderr to the defaults
+    # so that arg messages (i.e. --help) print out to terminal
+    original_stdout = sys.stdout
+    original_stderr = sys.stderr
+
+    try:
+        sys.stdout = sys.__stdout__
+        sys.stderr = sys.__stderr__
+
+        parser = argparse.ArgumentParser()
+
+        # Subparsers for 'I' and 'D'
+        subparsers = parser.add_subparsers(dest='method', required=True)
+        parser_I = subparsers.add_parser('I', help='Local file input')
+        parser_I.add_argument('--local-path',
+                            required=True,
+                            help='Path to the local GENDF file.')
+        parser_I.add_argument('--isomer', '-m',
+                              required=False,
+                              default=None,
+                              help = 'Isomeric state of the element')
+        parser_D = subparsers.add_parser('D', help='Download GENDF file')
+        parser_D.add_argument('--element', '-e',
+                            required=True,
+                            help= 'Chemical symbol for selected element (i.e. Ti).')
+        parser_D.add_argument('--A', '-a',
+                            required=True,
+                            help='Mass number for selected isotope (i.e. 48). If the target is an isomer, "m" after the mass number (i.e. 48m)')
+
+        args = parser.parse_args()
+        return args
+    
+    finally:
+        # Restore stdout and stderr to the logger
+        sys.stdout = original_stdout
+        sys.stderr = original_stderr
+
+# Define a function to read CSV files
+def read_csv(csv_path):
+    
+    # Initialize an empty dictionary
+    data_dict = {}
+
+    # Open the CSV file
+    with open(csv_path, mode='r') as file:
+        # Create a CSV reader object
+        csv_reader = csv.DictReader(file)
+
+        # Initialize the dictionary keys with empty lists
+        for column in csv_reader.fieldnames:
+            data_dict[column] = []
+
+        # Populate the dictionary with the CSV data
+        for row in csv_reader:
+            for column in csv_reader.fieldnames:
+                data_dict[column].append(row[column])
```

Comment:
I don't think this is the best data structure for this data.  When I look at how you use it, it seems you'd rather have a dictionary with key of the MT number and data of the other columns, esp. the 2nd column.

Data structure choice is important in python for a few reasons:
1. it prevents fragile data design - this one is theoretically fragile, but probably fine the way it is used
2. it allows for more readable code.

I think you want to translate these rows into a single dict with something like:
```
for row in csv_reader:
    data_dict[row['MT'] = row['reaction']
```